static/js/containers/pages/CheckoutPage_test.js

```diff
@@ -124,6 +124,39 @@ describe("CheckoutPage", () => {
   });
 
   it("submits the coupon code", async () => {});
+  it("submits the coupon code without trimming whitespace", async () => {
```

Comment:
We are duplicating a lot of code from the following test here. This is what I would suggest.

This is a parameterized test that accepts coupon codes as well. So, we can pass `xyz` and `xyz ` codes to that param along with existing ones. So either, had error is true of the coupon code is `xyz ` the field should show the error.